### PR TITLE
Slap people in the mouth with a pill bottle to feed them a random pill.

### PR DESCRIPTION
**Pull request by SonixApache**
Attempting to do this will first rattle the pill bottle, which is a one second do_after. After that it works identically to trying to feed someone a pill, chat message stating which pill you're using and all.

:cl:
 * rscadd: attacking someone with a pill bottle while targetting the mouth will now attempt to feed them a random pill from that bottle.

### CODE REVIEW

**Comment by wotisjanitor** [comment]
This slapping shit has gone too far

**Comment by GreatBigFailure** [comment]
As long as it's the same do_after() time as normal pills I don't see a reason not to have it. Functionally the same as just feeding the pill itself except add randomness. I guess the only difference is it would allow you to spam click and feed multiple pills, which could be pretty strong now that I think about it. Still seems fine to me.

**Comment by Dacendeth** [comment]
What about making it work with `/food/snacks/sweet`  so you can feed people random  candies too from the candybag which is really just a pillbottle.

**Comment by SonixApache** [comment]
> What about making it work with `/food/snacks/sweet` so you can feed people random candies too from the candybag which is really just a pillbottle.

i will consider it if i recover my mana, maybe I should just add support for papers and dice as well.

**Comment by Dacendeth** [comment]
I'm not sure about the mouth targeting requirement as otherwise It just slaps them with the pill bottle for 0 damage, is that interaction worth keeping at the cost of having an awkward/secret mouth targeting requirement for this interaction or is there something i'm missing?

**Comment by SonixApache** [comment]
no real reason other than "it made sense" , ensuring no accidental pill feeding.
Do you think I should just change it to the standard interaction, or green intent interaction, or what?

**Comment by Securitymain** [comment]
Why not the pill at the top of the stack? 
Why not all of them? 
Why have the rattle not only be a sound effect but also show up in text?
I second having do_after delay be the same as regular pills if you keep it as is. Define instead of magic number, surely pills have one?

I also fail to see the use case for this. If it's for quick application of chloral, harm intent mouth beaker. But then this doesn't have the mask/mouth check splashing has.
If it's for healing it is only useful if your entire bottle has the same pill, which is fine.

**Comment by GreatBigFailure** [comment]
> I also fail to see the use case for this.

Could be fun for things like force feeding people from the bottle, and only 50% of pills kill them, make a whole shtick of it

**Comment by falcon2346** [comment]
> I also fail to see the use case for this. If it's for quick application of chloral, harm intent mouth beaker. But then this doesn't have the mask/mouth check splashing has. If it's for healing it is only useful if your entire bottle has the same pill, which is fine.

I often have a single pill bottle filled with a single all purpose pill along with a more standard pill bottle. I think this could be really useful for bottles like that, or when you have a long lasting cocktail you've separated into multiple pills due to size limitations. Additionally, I'd bet you that if this gets merged you'll see a lot more bottles with only a single pill type on them just for convenience.

**Comment by falcon2346** [comment]
> no real reason other than "it made sense" , ensuring no accidental pill feeding. Do you think I should just change it to the standard interaction, or green intent interaction, or what?

I think this should just be a help intent interaction. You don't need to target the mouth to forcefeed someone, I see no reason to require the mouth to be targeted for this. It's not like there's any other reason to slap someone with a pill bottle.

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 1]
```diff
@@ -178,6 +178,24 @@ var/global/list/bottle_colour_choices = list("Blue" = "#0094FF","Dark Blue" = "#
...
+		pills.Add(pill)
+	if(!pills.len)
+		to_chat(user, "<span class='notice'>There are no pills left in \the [src].</span>")
+		return
+	var/obj/item/weapon/reagent_containers/pill/mypill = pick(pills)
+	to_chat(user, "<span class='notice'>You rattle \the [src] and [M == user ? "feed yourself" : "try to feed [M]"] with it.</span>")
+	playsound(src, 'sound/items/pillrattle.ogg', 10, 1, 1)
+	if(do_after(user, src, 1 SECONDS))
+		mypill.try_feed(M, user)
+	return
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/firstaid.dm:198]: no trailing return

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 2]
```diff
@@ -178,6 +178,24 @@ var/global/list/bottle_colour_choices = list("Blue" = "#0094FF","Dark Blue" = "#
 	overlays -= colour_overlay
 	colour_overlay.color = "[bottle_colour]"
 	overlays += colour_overlay
+	
+/obj/item/weapon/storage/pill_bottle/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/firstaid.dm:182]: get rid of "as mob" for both of these (convention)

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 3]
```diff
@@ -178,6 +178,24 @@ var/global/list/bottle_colour_choices = list("Blue" = "#0094FF","Dark Blue" = "#
...
+	if(!istype(M, /mob))
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/firstaid.dm:183]: style: !istype(M) is enough here (you already defined it to be a carbon above)

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 4]
```diff
@@ -178,6 +178,24 @@ var/global/list/bottle_colour_choices = list("Blue" = "#0094FF","Dark Blue" = "#
...
+	if (user.zone_sel.selecting != "mouth")
+		return ..()
+	var/list/pills = list()
+	for(var/obj/item/weapon/pill in contents)
+		pills.Add(pill)
+	if(!pills.len)
+		to_chat(user, "<span class='notice'>There are no pills left in \the [src].</span>")
+		return
+	var/obj/item/weapon/reagent_containers/pill/mypill = pick(pills)
+	to_chat(user, "<span class='notice'>You rattle \the [src] and [M == user ? "feed yourself" : "try to feed [M]"] with it.</span>")
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/firstaid.dm:194]: danger span for feeding other

File: code/game/objects/items/weapons/storage/firstaid.dm [hunk 5]
```diff
@@ -178,6 +178,24 @@ var/global/list/bottle_colour_choices = list("Blue" = "#0094FF","Dark Blue" = "#
...
+	playsound(src, 'sound/items/pillrattle.ogg', 10, 1, 1)
+	if(do_after(user, src, 1 SECONDS))
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/storage/firstaid.dm:196]: needs a do_after_many to check target remained in place

**Comment by SonixApache** [comment]
will address comments and review as soon as i get my work enviro on my new machine + the time to work on it

**Comment by SonixApache** [comment]
it's been three months and i still haven't set up git, maybe i'll get to it one day

**Pull request closed without merging**